command/acl_token_self.go

```diff
@@ -66,7 +66,17 @@ func (c *ACLTokenSelfCommand) Run(args []string) int {
 	// Get the specified token information
 	token, _, err := client.ACLTokens().Self(nil)
 	if err != nil {
-		c.Ui.Error(fmt.Sprintf("Error fetching self token: %s", err))
+		if !strings.Contains(err.Error(), "Unexpected response code: 404") {
```

Comment:
I think this bool check is reversed, because this is what happens when I use a non-existent token in current `main`:

```
$ NOMAD_TOKEN=10000000-0000-0000-0000-000000000000 nomad acl token self
Error fetching self token: Unexpected response code: 404 (ACL token not found)
```

```suggestion
		if strings.Contains(err.Error(), "Unexpected response code: 404") {
```

Reply:
🤔 I don't follow, Daniel. 

On `main` you will of course get 404 if you try to query `token self` with WI. What you're showing above is strange to me, prefixing `nomad acl token self` with a bogus token should give you 403 instead. 

The conditional here is correct, though. For any error _other_ than 404, we return error. In case we get 404, we further check job-attached policies, and only if we cannot find any, we say: no acl token and no policies attached. 

Does that make sense?

Reply:
what exactly I was thinking back then is lost to the mists of time, alas.

but, while I'm here comparing the two error messages from a bad, but valid (uuid), ACL token:

* nomad release 1.10 is clearly about one thing:
  > Error fetching self token: Unexpected response code: 404 (ACL token not found)

* this pr:
  > No ACL tokens or ACL policies attached to a workload identity found.

the new one seems to me like `acl token self` is only for workload ID, because I don't parse "or ACL policies attached to a workload identity" as a separate item? I get lost grammatically somewhere along the way before hitting "found" and I parse it as "No ... workload identity found."

this might be resolved for me with a single character:

> No ACL tokens `nor` ACL policies attached to a workload identity found.

for bonus clarity, if a bit choppy in rhythm:

> No ACL tokens, `nor` ACL policies attached to a workload identity, found.

Reply:
[56dfa5f](https://github.com/hashicorp/nomad/pull/25588/commits/56dfa5f88fc017cb6443666bf95d73ada4ecf640)